stl/inc/ranges

```diff
@@ -6470,20 +6470,33 @@ namespace ranges {
             constexpr _Iterator& operator+=(const difference_type _Off) requires random_access_range<_Base> {
                 if (_Off > 0) {
 #if _ITERATOR_DEBUG_LEVEL != 0
-                    _STL_VERIFY(_Off == 1 || _Stride <= (numeric_limits<difference_type>::max)() / (_Off - 1),
+                    _STL_VERIFY(_Stride <= (numeric_limits<difference_type>::max)() / _Off,
                         "cannot advance stride_view iterator past end (integer overflow)");
-                    _STL_VERIFY(_RANGES distance(_Current, _End) > _Stride * (_Off - 1),
-                        "cannot advance stride_view iterator past end");
+                    if constexpr (sized_sentinel_for<_Base_sentinel, _Base_iterator>) {
+                        _STL_VERIFY(_RANGES distance(_Current, _End) > _Stride * (_Off - 1),
+                            "cannot advance stride_view iterator past end");
+                    }
 #endif // _ITERATOR_DEBUG_LEVEL != 0
-                    _Missing = _RANGES advance(_Current, _Stride * _Off, _End);
+
+                    // Per to-be-filed LWG issue (See GH-2995)
+                    if constexpr (sized_sentinel_for<_Base_sentinel, _Base_iterator>) {
+                        _Missing = _RANGES advance(_Current, _Stride * _Off, _End);
+                    } else {
+                        _Current += _Stride * static_cast<difference_type>(_Off - 1);

```

Comment:
I observe that `chunk_view`'s iterator said `_Current += _Count * (_Off - 1);` without a `static_cast`, and the types involved appear to be the same here. No change requested, as the tests aren't emitting warnings.

Reply:
This is completely bogus, Casey: `*` promotes, too. Both of these need to convert immediately before adding to the iterator. (I'm not adding test coverage for this because I expect #2886 to do so.)